compiler/rustc_mir/src/interpret/cast.rs

```diff
@@ -142,6 +144,17 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         // Handle cast from a univariant (ZST) enum.
         match src.layout.variants {
             Variants::Single { index } => {
```

Comment:
@eddyb is it expected that uninhabited enums have `Variants::Single` layout?

Reply:
Answering myself: yes it is, `Single` doesn't mean "enum with 1 variant" but it means "layout with one variant" (as in, fields are not overlapping), and the only alternative is a more-than-1-variant layout (which can have overlapping fields), which certainly makes even less sense for uninhabited types.